docs/appendix.rst

```diff
@@ -524,76 +509,190 @@ To fix this, install the Python "wheel" package::
+
+
+.. _ImportError urllib3 v2.0 only supports OpenSSL 1.1.1+:
+
+ImportError urllib3 v2.0 only supports OpenSSL 1.1.1+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. index:: single LibreSSL
+.. index:: pair; troubleshooting: OpenSSL
+.. index:: single LibreSSL
+.. index:: LibreSSL macOS
+.. index:: single: TLS
+.. index:: pair; ImportError urllib3: OpenSSL
+
+The 'urllib3' Python package version 2.0 removed support for LibreSSL and
+some other SSL implementations and requires OpenSSL 1.1.1 or higher.  LibreSSL
+support was restored to 'urllib3' in version 2.0.3.
+
+The 'pywbem' package uses the 'requests' package which uses 'urllib3';
+neither 'pywbem' nor 'requests' pins 'urllib3' to stay below version 2.0.
+
+Therefore, upgrading Python packages, with a Python that does not support
+OpenSSL 1.1.1 or higher or other SSL implementations, causes the following
+exception raised by urllib3:
+
+    ImportError: urllib3 v2.0 only supports OpenSSL 1.1.1+, currently the ‘ssl’ module is compiled with LibreSSL 2.8.3.
+    See: https://github.com/urllib3/urllib3/issues/2168
+
+This can happen for example on macOS with the system Python of
+macOS as the basis for a Python virtual environment and installing pywbem
+into that virtual environment, which typically installs the latest available
+versions of dependent packages, and thus may install urllib3 with a version 2.0
+or later.
+
+The ImportError exception message shows the name and version of the underlying
+SSL library the Python 'ssl' module is using. On most Python systems, that is a
+statically linked SSL library, so just installing OpenSSL 1.1.1 or higher does
+not address the issue.
+
+Verify which SSL library and version as follows::
+
+  $ python -c "import ssl; print(ssl.OPENSSL_VERSION)"
+  OpenSSL 1.1.1t  7 Feb 2023
+
+  $ /usr/bin/python3 -c "import ssl; print(ssl.OPENSSL_VERSION)"
+  LibreSSL 2.8.3
+
+Note that Python since version 3.10 requires OpenSSL version 1.1.1 or higher
+(see PEP-644).
+
+At least up to macOS Ventura, Apple compiles the system Python with LibreSSL.
+As long as that does not change, you cannot use the system Python of macOS with
+urllib3>=2.0; also not as a basis for Python virtual environments.
+
+There are basically two options on how this issue can be addressed:
+
+* Use a Python version that uses OpenSSL 1.1.1 or higher. That is the case
+  for the CPython reference implementation version 3.7 or higher. CPython can
+  either be downloaded from https://www.python.org/downloads/macos/ or
+  installed using a third party package installer for macOS, such as Homebrew.
+
+* Pin the urllib3 package to stay below version 2.0 when on Python 3.7 or
+  higher, by specifying in the package dependencies, e.g. in the
+  requirements.txt file::
+
+    urllib3>=1.26.5,<2.0; python_version >= '3.7'
+
+The minimum version of urllib3 should be at least what the
+minimum-constraints.txt file of the 'pywbem' project specifies as a minimum,
+for the 'pywbem' version.
```

Comment:
For users of pywbem, it does not matter what we have in minimum-constraints.txt. What matters is what pywbem has in its package metadata, which comes from requirements.txt. I suggest not to talk about the requirements files at all, because it is simply the pywbem package requirements that matter, and pip will surface an error if it cannot match the requirements of the pywbem user with the requirements of pywbem.